### ISSUE TITLE
GitHub Workflow: support uploading build report from Windows

### ISSUE THREAD

**Issue by aSemy** [kotest/kotest#4092]
https://github.com/kotest/kotest/actions/runs/9482603753/job/26128051793

```
Run find . -type d -name 'reports' | zip -@ -r build-reports.zip
  find . -type d -name 'reports' | zip -@ -r build-reports.zip
  shell: C:\Program Files\PowerShell\7\pwsh.EXE -command ". '{0}'"
zip: D:\a\_temp\71c08419-1c19-4c[3](https://github.com/kotest/kotest/actions/runs/9482603753/job/26128051793#step:7:3)e-b0c6-fdbb3508ae53.ps1:2
Line |
   2 |  find . -type d -name 'reports' | zip -@ -r build-reports.zip
     |                                   ~~~
     | The term 'zip' is not recognized as a name of a cmdlet, function, script file, or executable program. Check the
     | spelling of the name, or if a path was included, verify that the path is correct and try again.
Error: Process completed with exit code 1.Run find . -type d -name 'reports' | zip -@ -r build-reports.zip
  find . -type d -name 'reports' | zip -@ -r build-reports.zip
  shell: C:\Program Files\PowerShell\7\pwsh.EXE -command ". '{0}'"
  env:
    RELEASE_VERSION: 
    OSSRH_USERNAME: 
    OSSRH_PASSWORD: 
    ORG_GRADLE_PROJECT_signingKey: 
    ORG_GRADLE_PROJECT_signingPassword: 
    GRADLE_OPTS: -Dorg.gradle.jvmargs="-Xmx1[2](https://github.com/kotest/kotest/actions/runs/9482603753/job/26128051793#step:7:2)g -XX:+HeapDumpOnOutOfMemoryError -Dfile.encoding=UTF-8"
    GRADLE_ACTION_ID: gradle/actions/setup-gradle
    JAVA_HOME: C:\hostedtoolcache\windows\Java_Temurin-Hotspot_jdk\17.0.11-9\x64
    JAVA_HOME_17_X64: C:\hostedtoolcache\windows\Java_Temurin-Hotspot_jdk\17.0.11-9\x64
    GRADLE_BUILD_ACTION_SETUP_COMPLETED: true
    GRADLE_BUILD_ACTION_CACHE_RESTORED: true
    DEVELOCITY_INJECTION_INIT_SCRIPT_NAME: gradle-actions.inject-develocity.init.gradle
    DEVELOCITY_AUTO_INJECTION_CUSTOM_VALUE: gradle-actions
    GITHUB_DEPENDENCY_GRAPH_ENABLED: false
zip: D:\a\_temp\71c08419-1c19-4c[3](https://github.com/kotest/kotest/actions/runs/9482603753/job/26128051793#step:7:3)e-b0c6-fdbb3508ae53.ps1:2
Line |
   2 |  find . -type d -name 'reports' | zip -@ -r build-reports.zip
     |                                   ~~~
     | The term 'zip' is not recognized as a name of a cmdlet, function, script file, or executable program. Check the
     | spelling of the name, or if a path was included, verify that the path is correct and try again.
Error: Process completed with exit code 1.Run find . -type d -name 'reports' | zip -@ -r build-reports.zip
  find . -type d -name 'reports' | zip -@ -r build-reports.zip
  shell: C:\Program Files\PowerShell\7\pwsh.EXE -command ". '{0}'"
  env:
    RELEASE_VERSION: 
    OSSRH_USERNAME: 
    OSSRH_PASSWORD: 
    ORG_GRADLE_PROJECT_signingKey: 
    ORG_GRADLE_PROJECT_signingPassword: 
    GRADLE_OPTS: -Dorg.gradle.jvmargs="-Xmx1[2](https://github.com/kotest/kotest/actions/runs/9482603753/job/26128051793#step:7:2)g -XX:+HeapDumpOnOutOfMemoryError -Dfile.encoding=UTF-8"
    GRADLE_ACTION_ID: gradle/actions/setup-gradle
    JAVA_HOME: C:\hostedtoolcache\windows\Java_Temurin-Hotspot_jdk\17.0.11-9\x64
    JAVA_HOME_17_X64: C:\hostedtoolcache\windows\Java_Temurin-Hotspot_jdk\17.0.11-9\x64
    GRADLE_BUILD_ACTION_SETUP_COMPLETED: true
    GRADLE_BUILD_ACTION_CACHE_RESTORED: true
    DEVELOCITY_INJECTION_INIT_SCRIPT_NAME: gradle-actions.inject-develocity.init.gradle
    DEVELOCITY_AUTO_INJECTION_CUSTOM_VALUE: gradle-actions
    GITHUB_DEPENDENCY_GRAPH_ENABLED: false
zip: D:\a\_temp\71c08419-1c19-4c[3](https://github.com/kotest/kotest/actions/runs/9482603753/job/26128051793#step:7:3)e-b0c6-fdbb3508ae53.ps1:2
Line |
   2 |  find . -type d -name 'reports' | zip -@ -r build-reports.zip
     |                                   ~~~
     | The term 'zip' is not recognized as a name of a cmdlet, function, script file, or executable program. Check the
     | spelling of the name, or if a path was included, verify that the path is correct and try again.
Error: Process completed with exit code 1.
```

Maybe set the shell to bash explicitly?

**Comment by OliverO2** [comment]
> Maybe set the shell to bash explicitly?

Done via 0e724fbbd01bcf0d51bd97769d8fffcad3c99b0f. Using a common shell is good practice anyway, but let's see whether this also puts a `zip` executable onto the `PATH`.

Now trying to fix Windows test runs in #4096.

**Comment by aSemy** [comment]
I think manually zipping isn't necessary. The upload-artifact action just zips everything. 

Example: https://github.com/adamko-dev/dokkatoo/blob/1ad48c5394b51fddb8d48f6061ca8168d5952d6b/.github/workflows/run_gradle_task.yml#L107-L116
